File: CS/OfficeNavigationBarCustomization.Module.Win/Controllers/OfficeNavigationBarCustomizationController.cs
using System;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Win.Templates;
using DevExpress.XtraBars.Ribbon;
using System.Windows.Forms;

namespace CustomizeOfficeNavigationBar.Win {
    public class OfficeNavigationBarCustomizationController : WindowController {
        private void Frame_TemplateChanged(object sender, EventArgs e) {
            Form form = Frame.Template as Form;
            if(form != null) {
                form.Load += Form_Load;
            }
        }
        private void Form_Load(object sender, EventArgs e) {
            IOfficeNavigationBarHolder officeNavigationBarHolder = sender as IOfficeNavigationBarHolder;
            if(officeNavigationBarHolder != null) {
                officeNavigationBarHolder.OfficeNavigationBar.MaxItemCount = 4;
            }
        }
        protected override void OnActivated() {
            base.OnActivated();
            Frame.TemplateChanged += Frame_TemplateChanged;
        }
        protected override void OnDeactivated() {
            Frame.TemplateChanged -= Frame_TemplateChanged;
            base.OnDeactivated();
        }

        public OfficeNavigationBarCustomizationController() {
            TargetWindowType = WindowType.Main;
        }
    }
}

File: VB/OfficeNavigationBarCustomization.Module.Win/Controllers/OfficeNavigationBarCustomizationController.vb
Imports System
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Win.Templates
Imports DevExpress.XtraBars.Ribbon

Namespace CustomizeOfficeNavigationBar.Win
    Public Class OfficeNavigationBarCustomizationController
        Inherits WindowController

        Private Sub Frame_TemplateChanged(ByVal sender As Object, ByVal e As EventArgs)
            Dim form As Form = TryCast(Frame.Template, Form)
            If form IsNot Nothing Then
                AddHandler form.Load, AddressOf Form_Load
            End If
        End Sub
        Private Sub Form_Load(ByVal sender As Object, ByVal e As EventArgs)
            Dim officeNavigationBarHolder As IOfficeNavigationBarHolder = TryCast(sender, IOfficeNavigationBarHolder)
            If officeNavigationBarHolder IsNot Nothing Then
                officeNavigationBarHolder.OfficeNavigationBar.MaxItemCount = 4
            End If
        End Sub
        Protected Overrides Sub OnActivated()
            MyBase.OnActivated()
            AddHandler Frame.TemplateChanged, AddressOf Frame_TemplateChanged
        End Sub
        Protected Overrides Sub OnDeactivated()
            RemoveHandler Frame.TemplateChanged, AddressOf Frame_TemplateChanged
            MyBase.OnDeactivated()
        End Sub

        Public Sub New()
            TargetWindowType = WindowType.Main
        End Sub
    End Class
End Namespace
